### PR TITLE
280 add custom data display on dashboard

### PR DESCRIPTION
**Pull request by Kwash67**

### CODE REVIEW

File: gcs/src/components/dashboardDataModal.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    Tooltip,
+  } from '@mantine/core'
+  import { IconInfoCircle, IconRefresh } from '@tabler/icons-react'
+  
+  // Styling imports
+  import resolveConfig from 'tailwindcss/resolveConfig'
+  import tailwindConfig from '../../tailwind.config.js'
+  
+  const tailwindColors = resolveConfig(tailwindConfig).theme.colors
+  const items = ['Item 1', 'Item 2', 'Item 3', 'Item 4', 'Item 5', 'Item 6', 'Item 7'];
```
Comment (1Blademaster) [inline gcs/src/components/dashboardDataModal.jsx:23]: Why 7 not 8?

File: gcs/src/components/dashboardDataModal.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      <Modal
+        opened={opened}
+        onClose={() => {
+          close()
+        }}
+        title='Select Data'
+        centered
+        overlayProps={{
+          backgroundOpacity: 0.55,
```
Comment (1Blademaster) [inline gcs/src/components/dashboardDataModal.jsx:39]: Is this the same background opacity for other modals throughout the app?

Comment (Kwash67) [reply gcs/src/components/dashboardDataModal.jsx:39]: yes

File: gcs/src/components/dashboardDataModal.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          blur: 3,
+        }}
+        withCloseButton={false}
+      >
+        {/* <LoadingOverlay visible={fetchingComPorts} /> */}
```
Comment (1Blademaster) [inline gcs/src/components/dashboardDataModal.jsx:44]: Is this meant to be commented out?

File: gcs/src/components/dashboardDataModal.jsx (hunk 1 above)
Comment (Kwash67) [reply gcs/src/components/dashboardDataModal.jsx:23]: just random😂. for testing

File: gcs/src/components/dashboardDataModal.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            Close
+          </Button>
+          <Button
+            variant='filled'
+            color={tailwindColors.green[600]}
+            onClick={() => close()}
+            data-autofocus
+          >
+            Confirm
+          </Button>
```
Comment (1Blademaster) [inline gcs/src/components/dashboardDataModal.jsx:70]: Both these buttons do the same thing?

File: gcs/src/components/dashboardDataModal.jsx (hunk 3 above)
Comment (Kwash67) [reply gcs/src/components/dashboardDataModal.jsx:44]: Oh, it's an oversight. I'll deal with that

File: gcs/src/dashboard.jsx [hunk 5]
```diff
@@ -123,6 +159,27 @@ export default function Dashboard() {
...
+  // Test Message Data 
+  const testData = {
+    "airspeed": 1.6184577941894531,
+    "groundspeed": 0.0274711474776268,
+    "heading": 227,
+    "throttle": 0,
+    "alt": 145.0500030517578,
+    "climb": 0.004107808228582144,
+    "timestamp": 1720022850.1995258
+  };
```
Comment (1Blademaster) [inline gcs/src/dashboard.jsx:177]: This shouldn't be in here right cuz it's test data?

File: gcs/src/dashboard.jsx [hunk 6]
```diff
@@ -151,6 +209,9 @@ export default function Dashboard() {
     socket.on('incoming_msg', (msg) => {
       if (incomingMessageHandler[msg.mavpackettype] !== undefined) {
         incomingMessageHandler[msg.mavpackettype](msg)
+        if(msg.mavpackettype in possibleData === false){
```
Comment (1Blademaster) [inline gcs/src/dashboard.jsx:212]: Can you add a comment explaining what this does pls

File: gcs/src/components/dashboardDataModal.jsx (hunk 4 above)
Comment (Kwash67) [reply gcs/src/components/dashboardDataModal.jsx:70]: I'll fix that

File: gcs/src/dashboard.jsx (hunk 6 above)
Comment (Kwash67) [reply gcs/src/dashboard.jsx:212]: okay

File: gcs/src/dashboard.jsx (hunk 5 above)
Comment (Kwash67) [reply gcs/src/dashboard.jsx:177]: Yeah, I'll clear it.

**Review by 1Blademaster — APPROVED** [review]

**Pull request merged**